### PR TITLE
Enumerable

### PR DESCRIPTION
**Pull request by husush23**
### In this PR we created `all`, `any`, and `filter` enumerables and you can check below the code implementation
We also created three more files for the new code as project per requirement

### CODE REVIEW

File: my_list.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  include MyEnumerable
+
+  def initialize(list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
+end
```
Comment (rotimiazeez) [inline my_list.rb:13]: Kindly ensure your enumerable is well created by running `ruby -r "./my_list.rb" -e "p MyList.new(1,2,4)"` 👍🏽

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi TEAM 👍🏽,
## Good job so far!
### To Highlight 👇🏽
- Good teamwork 💯 
- No Linter error ✔️ 
- Good PR description 👍🏽 
- Good commit message(s) 😉

You still need to work on some issues to prepare your project for the final evaluation, but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR WORDS>


## Suggested changes
- [ ] Please ensure the information provided in your README file matches the current project you are working on.
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag (@rotimiazeez) me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

File: my_list.rb (hunk 1 above)
Comment (husush23) [reply my_list.rb:13]: @rotimiazeez  Please can you make it more clear? what is the thing we are missing? what is the error you got in our code. The code works well for us, so let's get your suggestion clearly.

**Review by t-yanick — APPROVED** [review]
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @husush23 and @Katended , 

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Collaborative work :+1: Commits from both team members
:heavy_check_mark:  Good commit messages

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**